@ericnograles initializePushManager consist of two different registration
1. establish the connection with SFMC by `ETPush configureSDK:`
2. register remote notification to APNS

I think step 1 is per app launches, so if we try to call it again in the same app, it will throw an error like below.
(You can reproduce it by refreshing react-native view) 

```
RCTLog.js:38 Initializing ETPush with app ID XXX-XXX-XXX and access token XXXX
logErrorHandler.js:6 Error: Could not initialize JB4A-SDK with appId XXX-XXX-XXX and accesstoken XXXX. Please check your configuration.
-[ETPush configureSDKWithAppID:andAccessToken:withAnalytics:andLocationServices:andProximityServices:andCloudPages:withPIAnalytics:error:]: configureSDK for appId XXX-XXX-XXX has already completed.  You can only call configureSDK once.
    at createErrorFromErrorData (NativeModules.js:116)
    at NativeModules.js:79
    at MessageQueue.__invokeCallback (MessageQueue.js:288)
    at MessageQueue.js:127
    at guard (MessageQueue.js:46)
    at MessageQueue.invokeCallbackAndReturnFlushedQueue (MessageQueue.js:126)
    at debuggerWorker.js:71
```

Duplicate call of step 2 looks like no harm, but we can just filter it out not to do that by checking
`UIApplication.sharedApplication.registeredForRemoteNotifications`. This is Apple's standard.
https://developer.apple.com/documentation/uikit/uiapplication/1623069-registeredforremotenotifications?language=objc

--- a/index.js
+++ b/index.js
@@ -46,7 +46,21 @@ export default {
           });
       }
     );
+  },
 
+  registerForRemoteNotifications: () => {
+    return new Promise(
+      (resolve, reject) => {
+        RNExactTarget
+          .registerForRemoteNotifications()
+          .then(resolve)
+          .catch(reject)
+      }
+    );
+  },
+
+  setSubscriberKey: (key) => {
+    RNExactTarget.setSubscriberKey(key);
   },
 
   resetBadgeCount: () => {

--- a/ios/RNExactTarget.m
+++ b/ios/RNExactTarget.m
@@ -22,7 +22,7 @@
 
 bool hasListeners;
 
-bool isSdkInitialized = NO;
+bool hasSdkInitialized = NO;
 
 - (NSArray<NSString *> *)supportedEvents
 {
@@ -48,7 +48,7 @@ bool isSdkInitialized = NO;
 
 RCT_EXPORT_MODULE();
 
-RCT_REMAP_METHOD(registerForRemoteNotifications, registerForRemoteNotifications)
+RCT_REMAP_METHOD(registerForRemoteNotifications, registerForRemoteNotificationsWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
 {
     if (UIApplication.sharedApplication.registeredForRemoteNotifications) {
         RCTLogInfo(@"Remote notifications is already registered into APNS. We don't want to register it again.");
@@ -57,34 +57,40 @@ RCT_REMAP_METHOD(registerForRemoteNotifications, registerForRemoteNotifications)
     
     /** Register for push notifications - enable all notification types, no categories */
     RCTLogInfo(@"Registering for remote notifications");
-    if (floor(NSFoundationVersionNumber) > NSFoundationVersionNumber_iOS_9_x_Max) {
-        // Preparing parameters
-        UNAuthorizationOptions authOptions = (UNAuthorizationOptionAlert
-                                              + UNAuthorizationOptionBadge
-                                              + UNAuthorizationOptionSound);
-        void (^completionHandler)(BOOL, NSError * _Nullable) = ^(BOOL granted, NSError * _Nullable error) {
-            NSLog(@"Registered for remote notifications: %d", granted);
-        };
-        
-        // Start registration to APNS to get a device token
-        dispatch_async(dispatch_get_main_queue(), ^(void) {
-            [[ETPush pushManager] registerForRemoteNotificationsWithDelegate:self
-                                                                     options:authOptions
-                                                                  categories:nil
-                                                           completionHandler:completionHandler];
-        });
-    } else {
-        UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:
-                                                UIUserNotificationTypeBadge |
-                                                UIUserNotificationTypeSound |
-                                                UIUserNotificationTypeAlert
-                                                                                 categories:nil];
-        // Notify the SDK what user notification settings have been selected
-        dispatch_async(dispatch_get_main_queue(), ^(void) {
-            [[ETPush pushManager] registerUserNotificationSettings:settings];
-            [[ETPush pushManager] registerForRemoteNotifications];
-        });
+    @try {
+        if (floor(NSFoundationVersionNumber) > NSFoundationVersionNumber_iOS_9_x_Max) {
+            // Preparing parameters
+            UNAuthorizationOptions authOptions = (UNAuthorizationOptionAlert
+                                                  + UNAuthorizationOptionBadge
+                                                  + UNAuthorizationOptionSound);
+            void (^completionHandler)(BOOL, NSError * _Nullable) = ^(BOOL granted, NSError * _Nullable error) {
+                NSLog(@"Registered for remote notifications: %d", granted);
+            };
+            
+            // Start registration to APNS to get a device token
+            dispatch_async(dispatch_get_main_queue(), ^(void) {
+                [[ETPush pushManager] registerForRemoteNotificationsWithDelegate:self
+                                                                         options:authOptions
+                                                                      categories:nil
+                                                               completionHandler:completionHandler];
+            });
+        } else {
+            UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:
+                                                    UIUserNotificationTypeBadge |
+                                                    UIUserNotificationTypeSound |
+                                                    UIUserNotificationTypeAlert
+                                                                                     categories:nil];
+            // Notify the SDK what user notification settings have been selected
+            dispatch_async(dispatch_get_main_queue(), ^(void) {
+                [[ETPush pushManager] registerUserNotificationSettings:settings];
+                [[ETPush pushManager] registerForRemoteNotifications];
+            });
+        }
+    } @catch (NSException *exception) {
+        reject(@"apns_init_error", @"Error in registration for remote notifications", exception);
     }
+    
+    resolve(@"Registration for remote notifications fired");
 }
 
 RCT_REMAP_METHOD(initializePushManager, initializePushManager:(NSDictionary *)etPushConfig resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@@ -108,7 +114,7 @@ RCT_REMAP_METHOD(initializePushManager, initializePushManager:(NSDictionary *)et
 #endif
     
     // Initialize SDK with SFMC AppID and AccessToken
-    if (!isSdkInitialized) {
+    if (!hasSdkInitialized) {
         successful = [[ETPush pushManager] configureSDKWithAppID:appId
                                                   andAccessToken:accessToken
                                                    withAnalytics:withAnalytics
@@ -123,7 +129,7 @@ RCT_REMAP_METHOD(initializePushManager, initializePushManager:(NSDictionary *)et
             reject(@"sdk_init_error", errorMessage, error);
         }
         
-        isSdkInitialized = YES;
+        hasSdkInitialized = YES;
         RCTLogInfo(@"Initializing ETPush succeeded.", appId, accessToken);
     } else {
         RCTLogInfo(@"ETPush has already been initialized.", appId, accessToken);

